### PR TITLE
Create an AMM on Polkadot using Ink!

### PR DESCRIPTION
**Pull request by realnimish**
Thanks for submitting this tutorial for Figment Learn!

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://www.notion.so/figmentnetworks/Figment-Learn-Contributor-Guidelines-b08691366e144d33a5636d9276a44ec8) and reviewed the [Sample Tutorial Structure](https://www.notion.so/figmentnetworks/Sample-Tutorial-Structure-667ac2aad36f4f94a3ebaca053180b2d). 

Please confirm the following before submitting your tutorial for review - 

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://www.notion.so/figmentnetworks/Figment-Learn-Contributor-Guidelines-b08691366e144d33a5636d9276a44ec8).

In particular:

- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the [tutorial template](https://www.notion.so/figmentnetworks/Sample-Tutorial-Structure-667ac2aad36f4f94a3ebaca053180b2d) structure.
- [x] I've [proofread the tutorial](https://grammarly.com) for spelling and grammar mistakes.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run the tutorial to test it and the code works.

**NOTE**: We cannot review your Pull Request if it does not meet the standard outlined above.

### CODE REVIEW

File: polkadot/create-an-amm-on-polkadot-using-ink.md [hunk 1]
```diff
@@ -0,0 +1,806 @@
...
+
+```text
+git clone https://github.com/realnimish/polkadot-amm.git
+cd polkadot-amm
+npm install
+```
+
+In our react application we keep all the React components in the `src/components` directory.
+
+* **BoxTemplate** :-
```
Comment (pyr0gan) [inline polkadot/create-an-amm-on-polkadot-using-ink.md:731]: For all of these headings, you can remove the trailing -

> * **BoxTemplate** :

File: polkadot/create-an-amm-on-polkadot-using-ink.md [hunk 2]
```diff
@@ -0,0 +1,806 @@
...
+
+* **ContainerComponent** :- 
+This component renders the main body of our application which contains the center box, the tabs to switch between the five components Swap, Provide, Faucet, Withdraw, Account.
+
+The `App.js` renders the `ContainerComponent` and connects the application to `polkadot.{js}`. 
+
+The `constants.js` file stores the contract **ABI** and **CONTRACT_ADDRESS**. Don't forget to store your contract address and ABI in the respective variables. 
+
+{% hint style="info" %}  
+ABI can be obtained from your ink! project folder at `.../target/ink/metadata.json`
```
Comment (pyr0gan) [inline polkadot/create-an-amm-on-polkadot-using-ink.md:757]: > The ABI can be obtained from your ink! project folder at `/target/ink/metadata.json`

**Review by pyr0gan — CHANGES_REQUESTED** [review]
Hi @SayanKar and @realnimish !

Thanks for this well written and easy to follow tutorial. While there is quite a bit of code, you have done well with breaking it down and explaining both the AMM formula and the functions you're using to swap between tokens and the liquidity pool clearly 🙏 

There are some small issues to fix, please have a look over my comments and remember to mark the conversations as resolved when you make the changes 😄

File: polkadot/create-an-amm-on-polkadot-using-ink.md [hunk 3]
```diff
@@ -0,0 +1,806 @@
...
+}
+```
+
+## Part 4. Constructor
+
+Our constructor takes `_fees` as a parameter that determines the percent of fees the user is charged when doing a swap operation. The value of `_fees` should be between 0 and 1000(exclusive). Then any swap operation will be charged **_fees/1000** percent of the amount deposited.
+
+```rust
+/// Constructs a new AMM instance
+/// @param _fees: valid interval -> [0,1000)
```
[the remark this answers is not in the document]
Comment (realnimish) [reply polkadot/create-an-amm-on-polkadot-using-ink.md:156]: Hi @pyr0gan, The given brackets are intentional. Square brackets indicates closed end (inclusive) and open end (exclusive)

File: polkadot/create-an-amm-on-polkadot-using-ink.md [hunk 4]
```diff
@@ -0,0 +1,806 @@
...
+
+# How to interact with polkadot.{js}
+
+In this section, we will see how to interact with our smart contract using polkadot.{js}. Install the required packages 
+
+```text
+npm install @polkadot/api @polkadot/api-contract @polkadot/extension-dapp
+```
+
+Let's see the following code block, and understand how it works
```
[the remark this answers is not in the document]
Comment (SayanKar) [reply polkadot/create-an-amm-on-polkadot-using-ink.md:580]: The given section is just a template for developers to get an idea of how to interact with the smart contract from frontend.

**Review by pyr0gan — APPROVED** [review]
Changes look good, thanks gentlemen! I'll put this up on our staging for a final check before merging it this week 👍

**Pull request merged**